src/lib.rs

```diff
@@ -257,7 +257,7 @@ impl ClientOptions {
     #[cfg(feature = "ssl")]
     /// Creates a new options struct with a specified SSL certificate and key files.
     pub fn with_ssl(
-        ca_file: &str,
+        ca_file: Option<&str>,
```

Comment:
I realize that this is a breaking API change, but given the pre-1.0 version, it's technically semver compliant. While we could get around this by defining another pair of methods where the CA file isn't required, I think this one-time API change is worth not having four separate constructors for a relatively simple struct, given that the fix will be very trivial for users.